aws_lambda_builders/workflows/nodejs_npm_esbuild/workflow.py

```diff
@@ -53,150 +50,75 @@ def __init__(self, source_dir, artifacts_dir, scratch_dir, manifest_path, runtim
+                    osutils=self.osutils,
+                    subprocess_esbuild=self.subprocess_esbuild,
+                    manifest=self.manifest_path,
                 )
-            ]
+            )
             return
 
-        self.actions = self.actions_with_bundler(
-            source_dir, scratch_dir, artifacts_dir, bundler_config, osutils, subprocess_npm, subprocess_esbuild
-        )
-
-    def actions_with_bundler(
-        self, source_dir, scratch_dir, artifacts_dir, bundler_config, osutils, subprocess_npm, subprocess_esbuild
-    ) -> List[BaseAction]:
-        """
-        Generate a list of Nodejs build actions with a bundler
-
-        :type source_dir: str
-        :param source_dir: an existing (readable) directory containing source files
-
-        :type scratch_dir: str
-        :param scratch_dir: an existing (writable) directory for temporary files
-
-        :type artifacts_dir: str
-        :param artifacts_dir: an existing (writable) directory where to store the output.
-
-        :type bundler_config: dict
-        :param bundler_config: configurations for the bundler action
-
-        :type osutils: aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils
-        :param osutils: An instance of OS Utilities for file manipulation
-
-        :type subprocess_npm: aws_lambda_builders.workflows.nodejs_npm.npm.SubprocessNpm
-        :param subprocess_npm: An instance of the NPM process wrapper
-
-        :type subprocess_esbuild: aws_lambda_builders.workflows.nodejs_npm_esbuild.esbuild.SubprocessEsbuild
-        :param subprocess_esbuild: An instance of the esbuild process wrapper
-
-        :rtype: list
-        :return: List of build actions to execute
-        """
-        actions: List[BaseAction] = [
-            CopySourceAction(source_dir, scratch_dir, excludes=self.EXCLUDED_FILES + tuple(["node_modules"]))
-        ]
-
-        # Bundle dependencies separately in a dependency layer. We need to check the esbuild
-        # version here to ensure that it supports skipping dependency bundling
-        esbuild_no_deps = [
-            EsbuildCheckVersionAction(scratch_dir, subprocess_esbuild),
-            EsbuildBundleAction(
-                scratch_dir,
-                artifacts_dir,
-                bundler_config,
-                osutils,
-                subprocess_esbuild,
-                self.manifest_path,
-                skip_deps=True,
-            ),
-        ]
-        esbuild_with_deps = EsbuildBundleAction(
-            scratch_dir, artifacts_dir, bundler_config, osutils, subprocess_esbuild, self.manifest_path
-        )
+        if not self.download_dependencies and not self.dependencies_dir:
+            # Invalid workflow, can't have no dependency dir and no installation
+            raise EsbuildExecutionError(
+                message="Lambda Builders encountered an invalid workflow. A workflow can't "
+                "include a dependencies directory without installing dependencies."
```

Comment:
I just moved this around, didn't write the message myself, but I wanted to point out that it seems a bit unclear/misleading (the code comment above would be more accurate). 
The thing is that we need to either install dependencies or you need to provide us with a directory of where your dependencies are stored.

Reply:
That's right, we need to find the dependencies somewhere in this case. Suggested re-wording:
`Lambda Builders was unable to find the location of the dependencies since a dependencies directory was not provided and download dependencies is disabled.`

Reply:
done